doc/voq/voq_hld.md

```diff
@@ -0,0 +1,809 @@
+|:---:|:-----------:|:----------------------------------------------------------------------------------:|-----------------------------------|
+| 1.0 | 06/29/2020  | Sureshkannan Duraisamy, Srikanth Keesara, Vedavinayagam Ganesan (Nokia Sonic Team) | Initial public version            |
+|     |             |                                                                                    |                                   |
+
+# About this Manual
+This document describes the design details for supporting SONiC on a Distributed VOQ System. It aligns with the SONiC Distributed VOQ-Architecture and should be read in conjunction with that document. It also adopts the SONiC Multi-ASIC architecture is adopted - which allows each asic within an FSI to be controlled independently by a separate instance of the "SONiC Network Stack" (comprising bgp, swss, syncd, lldp, teamd etc ..).
+
+# Scope
+This specification is foussed primarily on IPv4/IPv6 unicast routing over Ethernet. No attempt is made in this specification to discuss how features like L2/L3 Multicast, Routing/Bridging over tunnels etc... might work with SONiC across a Distributed VOQ System. The expectation is that such features could be implemented in subsequent phases and would require additional work in SONiC and possibly additional SAI enhancements. But that is outside the scope of this document.
+
+# 1 Requirements Overview
+## 1.1 Functional Requirements
+
+The VOQ feature implementation shall support the following
+1.  Distributed VOQ System.
+2.  IPv4 and IPv6 unicast routing across any ports in the system
+3.  Each switch in the system is controlled by a separate "asic instance" of SONiC.
+4.  Host IP reachability to/from the interface IP addresses of any SONiC instance in the system using any network port in the system.
+5.  Host IP reachability between the SONiC instances in the system over the datapath.
+6.  Routing protocol peering between SONiC instances over the datapath.
+7.  Static provisioning of System Ports in the VOQ System Database.
+8.  Dynamic discovery of Routing Interfaces and Neighbors on other asics via the VOQ System Database.
+9.  Automatic determination of Switch_Id for each asic (optional).
+
+## 1.2 Platform requirements
+Every ASIC in the system needs be assigned a unique Switch_ID. Each ASIC consumes as many consecutive switch_id values as it has cores. So the next valid switch_id assignment should skip ahead by a number equal to the number of cores in the asic. This can be done via an additional configuration attribute. Another option is atuomatic determination of sWwitch_ID. Vendor supplied platform specific components are aware of details (line card number, asic number with in the line card, number of cores for the asic etc...) which allow them to automatically calculate the Switch_ID for each asic. For this reason it is proposed that the PMON could populate the Switch_ID for an asic into the APPL_DB.
```

Comment:
In a later section, the DEVICE_METADATA section carries the switch_id. Is the proposal still to have PMON populate the switch_id?